jac/jaclang/compiler/parser.py

```diff
@@ -627,9 +631,14 @@ def inherited_archs(self, kid: list[ast.AstNode]) -> ast.SubNodeList[ast.Expr]:
             inherited_archs: LT (atomic_chain COMMA)* atomic_chain GT
                            | COLON (atomic_chain COMMA)* atomic_chain COLON
             """
-            valid_inh = [i for i in kid if isinstance(i, ast.Expr)]
+            parents = []
+            self.match_token(Tok.LT) or self.match_token(Tok.COLON)
```

Comment:
this should be `self.match_token(Tok.LT) or self.consume_token(Tok.COLON)` same bellow